### DISCUSSION TITLE
Cant Install Longhorn, manager CrashLoopBackoff

### DISCUSSION THREAD

**Discussion by carta-at** [longhorn/longhorn#6357] [General]
Hello,

I tried to install longhorn to my k8s cluster via longhorn and via kubectl ... both methods are not working.
longhorn-manager-xxxx crashes ...

one example log from the manager pod ...
can anybody help me with this error ?

`
time="2023-07-20T07:28:48Z" level=info msg="Starting longhorn conversion webhook server"
W0720 07:28:48.770041       1 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2023-07-20T07:28:48Z" level=info msg="Waiting for conversion webhook to become ready"
time="2023-07-20T07:28:48Z" level=warning msg="Failed to get webhook health endpoint https://localhost:9501/v1/healthz" error="Get \"https://localhost:9501/v1/healthz\": dial tcp [::1]:9501: connect: connection refused"
time="2023-07-20T07:28:48Z" level=info msg="Active TLS secret longhorn-webhook-tls (ver=8474482) (count 2): map[listener.cattle.io/cn-longhorn-admission-webhook.longhor-59584d:longhorn-admission-webhook.longhorn-system.svc listener.cattle.io/cn-longhorn-conversion-webhook.longho-6a0089:longhorn-conversion-webhook.longhorn-system.svc listener.cattle.io/fingerprint:SHA1=2482227245860042D301EF9144FB154C658C5FFE]"
time="2023-07-20T07:28:48Z" level=info msg="Listening on :9501"
time="2023-07-20T07:28:48Z" level=info msg="Starting apiregistration.k8s.io/v1, Kind=APIService controller"
time="2023-07-20T07:28:48Z" level=info msg="Starting apiextensions.k8s.io/v1, Kind=CustomResourceDefinition controller"
time="2023-07-20T07:28:48Z" level=info msg="Starting /v1, Kind=Secret controller"
time="2023-07-20T07:28:48Z" level=info msg="Building conversion rules..."
time="2023-07-20T07:28:48Z" level=info msg="Updating TLS secret for longhorn-webhook-tls (count: 2): map[listener.cattle.io/cn-longhorn-admission-webhook.longhor-59584d:longhorn-admission-webhook.longhorn-system.svc listener.cattle.io/cn-longhorn-conversion-webhook.longho-6a0089:longhorn-conversion-webhook.longhorn-system.svc listener.cattle.io/fingerprint:SHA1=2482227245860042D301EF9144FB154C658C5FFE]"
time="2023-07-20T07:28:50Z" level=info msg="Webhook conversion is ready"
time="2023-07-20T07:28:50Z" level=warning msg="Started longhorn conversion webhook server"
time="2023-07-20T07:28:50Z" level=info msg="Starting longhorn admission webhook server"
W0720 07:28:50.772657       1 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2023-07-20T07:28:50Z" level=info msg="Waiting for admission webhook to become ready"
time="2023-07-20T07:28:50Z" level=warning msg="Failed to get webhook health endpoint https://localhost:9502/v1/healthz" error="Get \"https://localhost:9502/v1/healthz\": dial tcp [::1]:9502: connect: connection refused"
I0720 07:28:50.773851       1 shared_informer.go:311] Waiting for caches to sync for longhorn datastore
I0720 07:28:51.975421       1 request.go:696] Waited for 1.199945649s due to client-side throttling, not priority and fairness, request: GET:https://10.96.0.1:443/apis/longhorn.io/v1beta2/backingimages?limit=500&resourceVersion=0
time="2023-07-20T07:28:52Z" level=warning msg="Failed to get webhook health endpoint https://localhost:9502/v1/healthz" error="Get \"https://localhost:9502/v1/healthz\": dial tcp [::1]:9502: connect: connection refused"
I0720 07:28:53.074438       1 shared_informer.go:318] Caches are synced for longhorn datastore
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for nodes.longhorn.io (Node)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for settings.longhorn.io (Setting)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for recurringjobs.longhorn.io (RecurringJob)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for backingimages.longhorn.io (BackingImage)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for volumes.longhorn.io (Volume)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for orphans.longhorn.io (Orphan)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for snapshots.longhorn.io (Snapshot)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for supportbundles.longhorn.io (SupportBundle)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for systembackups.longhorn.io (SystemBackup)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for systemrestores.longhorn.io (SystemRestore)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for volumeattachments.longhorn.io (VolumeAttachment)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for engines.longhorn.io (Engine)"
time="2023-07-20T07:28:53Z" level=info msg="Add validaton handler for replicas.longhorn.io (Replica)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for backups.longhorn.io (Backup)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for backingimages.longhorn.io (BackingImage)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for backingImageManagers.longhorn.io (BackingImageManager)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for backingimagedatasources.longhorn.io (BackingImageDataSource)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for nodes.longhorn.io (Node)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for volumes.longhorn.io (Volume)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for engines.longhorn.io (Engine)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for recurringjobs.longhorn.io (RecurringJob)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for engineimages.longhorn.io (EngineImage)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for orphans.longhorn.io (Orphan)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for sharemanagers.longhorn.io (ShareManager)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for backupvolumes.longhorn.io (BackupVolume)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for snapshots.longhorn.io (Snapshot)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for replicas.longhorn.io (Replica)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for supportbundles.longhorn.io (SupportBundle)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for systembackups.longhorn.io (SystemBackup)"
time="2023-07-20T07:28:53Z" level=info msg="Add mutation handler for volumeattachments.longhorn.io (VolumeAttachment)"
time="2023-07-20T07:28:53Z" level=info msg="Active TLS secret longhorn-webhook-tls (ver=8474482) (count 2): map[listener.cattle.io/cn-longhorn-admission-webhook.longhor-59584d:longhorn-admission-webhook.longhorn-system.svc listener.cattle.io/cn-longhorn-conversion-webhook.longho-6a0089:longhorn-conversion-webhook.longhorn-system.svc listener.cattle.io/fingerprint:SHA1=2482227245860042D301EF9144FB154C658C5FFE]"
time="2023-07-20T07:28:53Z" level=info msg="Listening on :9502"
time="2023-07-20T07:28:53Z" level=info msg="Starting /v1, Kind=Secret controller"
time="2023-07-20T07:28:53Z" level=info msg="Building validation rules..."
time="2023-07-20T07:28:53Z" level=info msg="Starting apiregistration.k8s.io/v1, Kind=APIService controller"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:nodes Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc00087f520 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:settings Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc00066dd40 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=info msg="Starting apiextensions.k8s.io/v1, Kind=CustomResourceDefinition controller"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:recurringjobs Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc000647980 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:backingimages Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc000647e00 OperationTypes:[CREATE DELETE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:volumes Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0018abb00 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:orphans Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0004c5600 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:snapshots Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc000b05880 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:supportbundles Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc000b05dc0 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:systembackups Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc00186aa80 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:systemrestores Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0004c5ce0 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:volumeattachments Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc000c28500 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:engines Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc000005b00 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:replicas Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc00072b600 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=info msg="Building mutation rules..."
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:backups Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0000e9180 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:backingimages Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0013f4600 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:backingImageManagers Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc00186b500 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:backingimagedatasources Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0012f1a00 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:nodes Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc00087fd40 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:volumes Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc000d0d680 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:engines Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc001502900 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:recurringjobs Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0013f4a80 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:engineimages Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc00082f2c0 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:orphans Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc001894160 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:sharemanagers Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0018946e0 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:backupvolumes Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc00132a000 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:snapshots Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0004c61c0 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:replicas Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc001898b00 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:supportbundles Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0004c6700 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:systembackups Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc0004c6c40 OperationTypes:[CREATE]}"
time="2023-07-20T07:28:53Z" level=debug msg="Add rule for {Name:volumeattachments Scope:Namespaced APIGroup:longhorn.io APIVersion:v1beta2 ObjectType:0xc000c288c0 OperationTypes:[CREATE UPDATE]}"
time="2023-07-20T07:28:53Z" level=info msg="Updating TLS secret for longhorn-webhook-tls (count: 2): map[listener.cattle.io/cn-longhorn-admission-webhook.longhor-59584d:longhorn-admission-webhook.longhorn-system.svc listener.cattle.io/cn-longhorn-conversion-webhook.longho-6a0089:longhorn-conversion-webhook.longhorn-system.svc listener.cattle.io/fingerprint:SHA1=2482227245860042D301EF9144FB154C658C5FFE]"
time="2023-07-20T07:28:53Z" level=debug msg="DesiredSet - No change(2) admissionregistration.k8s.io/v1, Kind=ValidatingWebhookConfiguration /longhorn-webhook-validator for  longhorn-system/longhorn-webhook-ca"
time="2023-07-20T07:28:53Z" level=debug msg="DesiredSet - No change(2) admissionregistration.k8s.io/v1, Kind=MutatingWebhookConfiguration /longhorn-webhook-mutator for  longhorn-system/longhorn-webhook-ca"
time="2023-07-20T07:28:54Z" level=info msg="Webhook admission is ready"
time="2023-07-20T07:28:54Z" level=warning msg="Started longhorn admission webhook server"
time="2023-07-20T07:28:54Z" level=info msg="Starting longhorn recovery-backend server"
W0720 07:28:54.776091       1 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0720 07:28:54.777123       1 shared_informer.go:311] Waiting for caches to sync for longhorn datastore
I0720 07:28:57.077769       1 shared_informer.go:318] Caches are synced for longhorn datastore
time="2023-07-20T07:28:57Z" level=info msg="Starting apiregistration.k8s.io/v1, Kind=APIService controller"
time="2023-07-20T07:28:57Z" level=info msg="Starting apiextensions.k8s.io/v1, Kind=CustomResourceDefinition controller"
time="2023-07-20T07:28:57Z" level=info msg="Started longhorn recovery-backend server"
W0720 07:28:57.187447       1 client_config.go:618] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2023-07-20T07:28:57Z" level=info msg="Recovery-backend server is running at :9503"
I0720 07:28:57.193863       1 leaderelection.go:245] attempting to acquire leader lease longhorn-system/longhorn-manager-upgrade-lock...
I0720 07:28:57.201236       1 leaderelection.go:255] successfully acquired lease longhorn-system/longhorn-manager-upgrade-lock
time="2023-07-20T07:28:57Z" level=info msg="Start upgrading"
time="2023-07-20T07:28:57Z" level=info msg="No API version upgrade is needed"
time="2023-07-20T07:28:57Z" level=info msg="Walking through the resource upgrade path v1.4.x to v1.5.0"
time="2023-07-20T07:28:57Z" level=debug msg="admit result: UPDATE longhorn.io/v1beta2, Kind=Node longhorn-system/svr-k8sw001 user=system:serviceaccount:longhorn-system:longhorn-service-account allowed=true err=<nil>"
time="2023-07-20T07:28:57Z" level=warning msg="Rejected operation: Request (user: system:serviceaccount:longhorn-system:longhorn-service-account, longhorn.io/v1beta2, Kind=Node, namespace: longhorn-system, name: svr-k8sw001, operation: UPDATE)" error="spec and status of disks on node svr-k8sw001 are being syncing and please retry later." service=admissionWebhook
time="2023-07-20T07:28:57Z" level=debug msg="admit result: UPDATE longhorn.io/v1beta2, Kind=Node longhorn-system/svr-k8sw001 user=system:serviceaccount:longhorn-system:longhorn-service-account allowed=false err=<nil>"
time="2023-07-20T07:28:57Z" level=error msg="Upgrade failed: upgrade resources failed: admission webhook \"validator.longhorn.io\" denied the request: spec and status of disks on node svr-k8sw001 are being syncing and please retry later."
time="2023-07-20T07:28:57Z" level=info msg="Upgrade leader lost: svr-k8sw002"
time="2023-07-20T07:28:57Z" level=fatal msg="Error starting manager: upgrade resources failed: admission webhook \"validator.longhorn.io\" denied the request: spec and status of disks on node svr-k8sw001 are being syncing and please retry later."
`

**Comment by mantissahz** [comment]
@carta-at
Could you describe more about you underlying infrastructure?
Which Longhorn version did you try to install? It seems that you are upgrading from v1.4.x
Kubernetes version, OS, VM or bare-mental?

**Comment by carta-at** [comment]
hi,
lognhorn 1.5.1
fresh install no upgrade ... tried to install 1.4.x 2 weeks ago same error

`
svr-k8sm001   Ready    control-plane   36d   v1.27.3   10.110.0.31   <none>        Ubuntu 22.04.2 LTS   5.15.0-76-generic   containerd://1.6.21
svr-k8sm002   Ready    control-plane   36d   v1.27.3   10.110.0.32   <none>        Ubuntu 22.04.2 LTS   5.15.0-76-generic   containerd://1.6.21
svr-k8sm003   Ready    control-plane   36d   v1.27.3   10.110.0.33   <none>        Ubuntu 22.04.2 LTS   5.15.0-76-generic   containerd://1.6.21
svr-k8sw001   Ready    worker          22d   v1.27.3   10.110.0.51   <none>        Ubuntu 22.04.2 LTS   5.15.0-76-generic   containerd://1.6.21
svr-k8sw002   Ready    worker          21d   v1.27.3   10.110.0.52   <none>        Ubuntu 22.04.2 LTS   5.15.0-76-generic   containerd://1.6.21
svr-k8sw003   Ready    worker          21d   v1.27.3   10.110.0.53   <none>        Ubuntu 22.04.2 LTS   5.15.0-76-generic   containerd://1.6.21
svr-k8sw004   Ready    worker          21d   v1.27.3   10.110.0.54   <none>        Ubuntu 22.04.2 LTS   5.15.0-76-generic   containerd://1.6.21
`
4x worker nodes are bare metal
3x control plan nodes are vms

os on all nodes :
Ubuntu 22.04.2 LTS
k8s version:
v1.27.3
container runtime:
containerd://1.6.21

**Comment by innobead** [comment]
> node svr-k8sw001 are being syncing and please retry later." time="2023-07-20T07:28:57Z" level=info msg="Upgrade leader lost: svr-k8sw002" time="2023-07-20T07:28:57Z" level=fatal msg="Error starting manager: upgrade resources failed: admission webhook \"validator.longhorn.io\" denied the request: spec and status of disks on node svr-k8sw001 are being syncing and please retry later."

@carta-at is your error still as above? It just means node spec and status are not synced correctly, could you provide the support bundle here? @mantissahz Please follow up. 

[/webhook/resources/node/validator.go#L81-L84](https://github.com/longhorn/longhorn-manager/blob/b6fd84b97d0bd2380c1bf0a31710dabc574fadc0/webhook/resources/node/validator.go#L81-L84)
```go
	// Ensure the node controller already syncs the disk spec and status.
	if !isNodeDiskSpecAndStatusSynced(oldNode) {
		return werror.NewForbiddenError(fmt.Sprintf("spec and status of disks on node %v are being syncing and please retry later.", oldNode.Name))
	}
```

**Reply by carta-at** [reply]
`could you provide the support bundle here`
what do i have to do ?

**Reply by mantissahz** [reply]
https://longhorn.io/docs/1.5.1/advanced-resources/support-bundle/
generate the support bundle, and you can attach the file here or send to longhorn-support-bundle@Suse.com

**Reply by derekbit** [reply]
@carta-at 
I think you can check the `nodes.longhorn.io` resources and see why the disks are not ready.

**Reply by carta-at** [reply]
```
Conditions:
    Last Transition Time:  2023-07-20T11:41:24Z
    Message:               no conflicts found
    Reason:                NoConflicts
    Status:                True
    Type:                  NamesAccepted
    Last Transition Time:  2023-07-20T11:41:24Z
    Message:               the initial names have been accepted
    Reason:                InitialNamesAccepted
    Status:                True
    Type:                  Established
```

**Reply by derekbit** [reply]
Can you provide the results of `kubectl -n longhorn-system get nodes.longhorn.io -o yaml`?

**Reply by carta-at** [reply]
```
apiVersion: v1
items:
- apiVersion: longhorn.io/v1beta2
  kind: Node
  metadata:
    creationTimestamp: "2023-07-04T11:00:59Z"
    deletionGracePeriodSeconds: 0
    deletionTimestamp: "2023-07-04T11:14:05Z"
    finalizers:
    - longhorn.io
    generation: 2
    name: svr-k8sw001
    namespace: longhorn-system
    resourceVersion: "8244718"
    uid: 91cd2025-394f-4b8b-ab96-e9efac1d969f
  spec:
    allowScheduling: true
    disks:
      default-disk-dd8737ad91178e18:
        allowScheduling: true
        evictionRequested: false
        path: /var/lib/longhorn/
        storageReserved: 67000570675
        tags: []
    evictionRequested: false
    name: svr-k8sw001
    tags: []
  status:
    conditions:
    - lastProbeTime: ""
      lastTransitionTime: "2023-07-04T11:00:59Z"
      message: 'Node svr-k8sw001 is down: the manager pod longhorn-manager-qzsch is
        not running'
      reason: ManagerPodDown
      status: "False"
      type: Ready
    - lastProbeTime: ""
      lastTransitionTime: "2023-07-04T11:00:59Z"
      message: ""
      reason: ""
      status: "True"
      type: Schedulable
    diskStatus: null
    region: ""
    snapshotCheckStatus: {}
    zone: ""
- apiVersion: longhorn.io/v1beta2
  kind: Node
  metadata:
    creationTimestamp: "2023-07-04T11:00:59Z"
    deletionGracePeriodSeconds: 0
    deletionTimestamp: "2023-07-04T11:14:05Z"
    finalizers:
    - longhorn.io
    generation: 2
    name: svr-k8sw004
    namespace: longhorn-system
    resourceVersion: "8244719"
    uid: dd9bff0d-028b-431e-aea7-1aae74a11d7b
  spec:
    allowScheduling: true
    disks:
      default-disk-6f954202221cc5eb:
        allowScheduling: true
        evictionRequested: false
        path: /var/lib/longhorn/
        storageReserved: 67000570675
        tags: []
    evictionRequested: false
    name: svr-k8sw004
    tags: []
  status:
    conditions:
    - lastProbeTime: ""
      lastTransitionTime: "2023-07-04T11:00:59Z"
      message: 'Node svr-k8sw004 is down: the manager pod longhorn-manager-r7cf6 is
        not running'
      reason: ManagerPodDown
      status: "False"
      type: Ready
    - lastProbeTime: ""
      lastTransitionTime: "2023-07-04T11:00:59Z"
      message: ""
      reason: ""
      status: "True"
      type: Schedulable
    diskStatus: null
    region: ""
    snapshotCheckStatus: {}
    zone: ""
kind: List
metadata:
  resourceVersion: ""
```

**Reply by carta-at** [reply]
ok so now i am a step further ... i edited the values.yaml of the helm chart and set the default data path to /mnt/data where my ssd is mounted
so now the sync error is gone .. but i have a different error:
```
W0721 08:38:23.753464       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2023-07-21T08:38:23Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/*], output , stderr ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/*': No such file or directory\n: exit status 2"
I0721 08:38:23.767855       1 leaderelection.go:248] attempting to acquire leader lease longhorn-system/longhorn-manager-upgrade-lock...
I0721 08:38:23.776980       1 leaderelection.go:258] successfully acquired lease longhorn-system/longhorn-manager-upgrade-lock
time="2023-07-21T08:38:23Z" level=info msg="Start upgrading"
time="2023-07-21T08:38:23Z" level=info msg="setting default-engine-image not found"
time="2023-07-21T08:38:23Z" level=error msg="Upgrade failed: upgrade API version failed: cannot create CRDAPIVersionSetting: Internal error occurred: failed calling webhook \"validator.longhorn.io\": failed to call webhook: Post \"https://longhorn-admission-webhook.longhorn-system.svc:9502/v1/webhook/validaton?timeout=10s\": dial tcp 10.99.128.13:9502: connect: connection refused"       
time="2023-07-21T08:38:23Z" level=info msg="Upgrade leader lost: svr-k8sw002"
time="2023-07-21T08:38:23Z" level=fatal msg="Error starting manager: upgrade API version failed: cannot create CRDAPIVersionSetting: Internal error occurred: failed calling webhook \"validator.longhorn.io\": failed to call webhook: Post \"https://longhorn-admission-webhook.longhorn-system.svc:9502/v1/webhook/validaton?timeout=10s\": dial tcp 10.99.128.13:9502: connect: connection refused"
```

**Reply by carta-at** [reply]
ok, changed in the values yaml the container versions and made the upgrade
now we are back to
level=fatal msg="Error starting manager: upgrade resources failed: admission webhook \"validator.longhorn.io\" denied the request: spec and status of disks on node svr-k8sw001 are being syncing and please retry later."

**Reply by mantissahz** [reply]
Because objects of `nodes.longhorn.io` will never be updated if `longhorn manager` pods are not finishing the upgrade procedure to sync the status. 
It seems that you did not clean up the Longhorn v1.4.x (it should have something wrong) and then install the v1.5.1 (this would be an upgrade.)
Suggest that you clean up Longhorn you installed and install Longhorn from scratch

**Reply by carta-at** [reply]
i did try to uninstall ...via helm it fails 
so i had to delete via kubectl .. and delete the ns and cleanup finalizers to get the NS delted ...
after that a new install leads always to my upgrade issue ??

is there a guide for manual cleanup ?

**Reply by carta-at** [reply]
little update ... i did manage to install 1.4.1 via helm without a problem ...
will try to update tomorrow to 1.5.x .. weired issue ...

**Reply by carta-at** [reply]
i might have found the cause ....
1.4.2 is working but only 2 nodes ...
the nodes that are shown as down have an age from x days ago where i first tried to install 
```
kubectl api-resources --verbs=list --namespaced -o name | xargs -n 1 kubectl get --show-kind --ignore-not-found -n longhorn-system
NAME                           READY   ALLOWSCHEDULING   SCHEDULABLE   AGE
node.longhorn.io/svr-k8sw001   False   true              True          17d
node.longhorn.io/svr-k8sw004   False   true              True          17d
``` 
any idea how to delete that for good ?
```
kubectl delete node.longhorn.io/svr-k8sw001 -n longhorn-system
``` 
is not working ...

**Reply by mantissahz** [reply]
Is your Longhorn version 1.4.2 now? Is `webhook` server or `longhorn-manager` running?
Maybe you could remove the finalizer `longhorn.io` in the object `node.longhorn.io/svr-k8sw001` then you could remove it.
